Loop/View Models/BolusEntryViewModel.swift

```diff
@@ -409,8 +486,13 @@ final class BolusEntryViewModel: ObservableObject {
     private func updatePredictedGlucoseValues(from state: LoopState, completion: @escaping () -> Void = {}) {
         dispatchPrecondition(condition: .notOnQueue(.main))
 
-        let (manualGlucoseSample, enteredBolus) = DispatchQueue.main.sync { (self.manualGlucoseSample, self.enteredBolus) }
-        let enteredBolusDose = DoseEntry(type: .bolus, startDate: Date(), value: enteredBolus.doubleValue(for: .internationalUnit()), unit: .units)
+        let (manualGlucoseSample, enteredBolus, doseDate, insulinModel) = DispatchQueue.main.sync { (self.manualGlucoseSample, self.enteredBolus, self.selectedDoseDate, self.selectedInsulinModel) }
+        var insulinModelSettings: InsulinModelSettings?
+        if let model = insulinModel {
+            insulinModelSettings = InsulinModelSettings(model: model)
+        }
```

Comment:
We only add the insulin model if it's actually set (which is not the case for manual boluses)